src/iterators.rs

```diff
@@ -618,6 +621,24 @@ macro_rules! custom_vec_iter_impl {
                 }
             }
 
+            #[classmethod]
+            #[pyo3(signature = (key, /))]
+            pub fn __class_getitem__(
+                cls: &Bound<'_, pyo3::types::PyType>,
+                key: &Bound<'_, PyAny>,
+            ) -> PyResult<PyObject> {
+                let class_string = stringify!($name);
+                const ALLOWED_GENERIC_CLASSES: [&str; 3] =
```

Comment:
Out of all changes this is the one I am least confident about because it feels like a hack